Urban-Fun/View Controllers/SearchViewController.m

```diff
@@ -6,26 +6,225 @@
 //
 
 #import "SearchViewController.h"
+#import "Activity.h"
+#import "User.h"
+#import "TimelineCell.h"
+#import "ProfileViewController.h"
+#import "ActivityDetailsViewController.h"
 
-@interface SearchViewController ()
-
+@interface SearchViewController () <UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, CLLocationManagerDelegate, TimelineCellDelegate, ActivityDetailsDelegate>
+@property (strong, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) IBOutlet UISearchBar *searchBar;
+@property (strong, nonatomic) IBOutlet UIBarButtonItem *filtersButton;
+@property (strong, nonatomic) IBOutlet UISegmentedControl *searchControl;
+@property (nonatomic, strong) NSMutableArray<User *> *usersArray;
+@property (nonatomic, strong) NSMutableArray<Activity *> *activitiesArray;
+@property (nonatomic, strong) User *profileToView;
+@property (nonatomic, strong) const CLLocationManager *manager;
+@property (nonatomic, strong) CLLocation *currentUserLocation;
 @end
 
 @implementation SearchViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.searchBar.delegate = self;
+    
+    // dismisses keyboard when tap outside a text field
+    UITapGestureRecognizer *tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self.view action:@selector(endEditing:)];
+    [tapGestureRecognizer setCancelsTouchesInView:NO];
+    [self.view addGestureRecognizer:tapGestureRecognizer];
+    // dismisses keyboard when user scrolls
+    self.tableView.keyboardDismissMode = UIScrollViewKeyboardDismissModeOnDrag;
+    
     // Do any additional setup after loading the view.
+    UIFont *font = [UIFont fontWithName:@"Verdana-Bold" size:12.0f];
```

Comment:
I'd suggest to change this name variables to `selectedColor` then you can use `color` for the normal state rather than `color2`